### ISSUE TITLE
Connect `mesher` to `elements`

### ISSUE THREAD

**Issue by sinaatalay** [fastfem/fastfem#24]
Hi @int-ptr-ptr. I’ve made some changes, but they aren’t exactly as you requested. I thought it would make sense to store elements and nodes in a dictionary, with the keys storing tags and the values storing coordinates for nodes or node tags for elements.

If this structure makes it difficult to interact with, please let me know, and I can change it.

Also, if you need any help with the integration, I would be happy to help. Perhaps I can write the code to translate my mesh to fit your interface.

**Comment by int-ptr-ptr** [comment]
It is certainly workable, but I worry about the inefficiencies of using a map to index nodes. I would expect that `mesh.nodes` would always be dense (contain all of the indices from `1` to `num_nodes`). If we re-index all nodes so that they are zero-indexed, then one can store all of the nodes in one large `ndarray`, allowing for vectorized access, say `top_bdry_node_coords = mesh.nodes[np.array(mesh["top_boundary"].mesh[0].nodes.keys()),:]`. The main thing that I am looking for is vectorization, but I think the memory overhead of having a separate coordinate array for each node is also notable.

If I were to integrate the mesher into elements, the first thing I would do would be to copy all of the data into full `numpy` arrays, so it would make sense to just have that in the base mesher already. If I was going to generate the position field from a mesh, I would do it with something like:
```python
pos_field_coefs = base_mesh.nodes[surface_mesh.elements.T, :2]
```


It also looks like the position data is copied twice. If I modify the node coordinates from the submesh, it does not affect the main mesh node dictionary, which tells me there are two copies of the data.
```python
tmp = mesh["top_boundary"].mesh[0].nodes[3][0] # store the old position temporarily
mesh["top_boundary"].mesh[0].nodes[3][0] = -1 # modify the submesh node coord
print(f"submesh node 3: {mesh['top_boundary'].mesh[0].nodes[3]}")
print(f"mesh node 3: {mesh.nodes[3]}")
mesh["top_boundary"].mesh[0].nodes[3][0] = tmp # restore the old position
# ========= OUTPUT ===========
submesh node 3: [-1.  1.  0.]
mesh node 3: [1. 1. 0.]
```

**Comment by sacha-escudier** [comment]
I initially thought keeping the structure as a dictionary might be preferable; it seems to be a more organic way to work with PyVista, since it requires both the coordinates of nodes and strips as its main arguments. But, I will try out the array appraoch you laid out here.

**Comment by int-ptr-ptr** [comment]
I think you could do both. If the dictionary works best for PyVista, you could override a `__get_item__` dunder in some class somewhere, or you could just make the dictionary with slice subviews.

**Comment by int-ptr-ptr** [comment]
> If we re-index all nodes so that they are zero-indexed, then one can store all of the nodes in one large `ndarray`, allowing for vectorized access...

@sinaatalay are we doing this or keeping the dictionary? I can review #13 if we're keeping it as-is.

**Comment by sinaatalay** [comment]
I will do it. For now, can you manually put all the dictionary values into an array? It should be a line of code.

I think PR is good to review; it will be just a couple of lines of change. I will do it in my first opportunity.

**Comment by int-ptr-ptr** [comment]
This issue should be resolved by #33.
